### PR TITLE
refactor: add lint-staged and tests run to git-hooks

### PR DESCRIPTION
**Pull request by thabarbados**
### Fix or Enhancement?
Add simple-git-hooks and config it. Add lint-staged for pre-commit hook

- [x] All tests passed

### Environment
- OS: macOs BigSur
- NPM Version: 7.21.1

### CODE REVIEW

File: .lintstagedrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "*.js": ["eslint", "prettier --write"]
```
Comment (dangreen) [inline .lintstagedrc:2]: Let's add `.ts` files for the future. Also, what about `.vue` files? I see this extension in the lint script.

**Review by dangreen — CHANGES_REQUESTED** [review]

File: .lintstagedrc (hunk 1 above)
Comment (thabarbados) [reply .lintstagedrc]: add checks for `.ts` and `.vue`

File: .simple-git-hooks.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "pre-commit": "yarn lint-staged",
```
Comment (dangreen) [inline .simple-git-hooks.json:2]:
```suggestion
{
  "commit-msg": "yarn commitlint --edit \"$1\"",
  "pre-commit": "yarn lint-staged",
```

**Review by dangreen — CHANGES_REQUESTED** [review]

File: .simple-git-hooks.json (hunk 2 above)
Comment (thabarbados) [reply .simple-git-hooks.json]: fix

**Review by dangreen — APPROVED** [review]

**Pull request merged**